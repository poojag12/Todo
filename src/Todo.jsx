import React, {useState, useEffect} from 'react';
import Form from "./Form";

export function hello() {
    console.log("Hello World");
}

function Todo(){
    const [todos, setTodo] = useState([]);
    const [currentTodoText, addTodo] = useState("");

    useEffect(() => {
        hello();
    }, []);
    const addTodoClick = (e) => {
       const newTodo = [...todos, currentTodoText];
        e.preventDefault();
        setTodo(newTodo);
        addTodo("")
    };

    return (
        <div className="App">
            <h1>ToDo</h1>
            <div>
                <Form addTodo={addTodo} addTodoClick={addTodoClick}
                      currentValue={currentTodoText}/>
                <ul className="todo-list">{todos.map(todo => <h1 key={todo}>{todo}</h1>)}</ul>
            </div>
        </div>
    );
}

export default Todo;
