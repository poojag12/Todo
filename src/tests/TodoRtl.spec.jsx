import React from "react";
import {render, fireEvent} from "@testing-library/react";
import Todo from "../Todo";

describe('Todo', () => {
    it('should render todo Todo ', () => {
        const {asFragment} = render(<Todo/>);
        expect(asFragment()).toMatchSnapshot();
    });

    it('should set new todo item in state', () => {
        const {getByText, container} = render(<Todo/>);

        const todoInput = container.querySelector("input");
        fireEvent.change(todoInput, {target: {value: 'Read book'}});

        const addTodo =  container.querySelector("button");
        fireEvent.click(addTodo, {preventDefault: jest.fn()});

        const todoItem = getByText('Read book');
        expect(todoItem).toBeDefined();
    });
});