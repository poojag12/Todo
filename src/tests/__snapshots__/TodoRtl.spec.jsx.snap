// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo should render todo Todo  1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      ToDo
    </h1>
    <div>
      <form
        class="form"
      >
        <input
          type="text"
          value=""
        />
        <button
          class="btn-primary"
          type="submit"
        >
          Add
        </button>
      </form>
      <ul
        class="todo-list"
      />
    </div>
  </div>
</DocumentFragment>
`;
